### PR TITLE
fix: pass same info to onClientUploadComplete as onUploadComplete

### PR DESCRIPTION
**Pull request by markflorkowski**
Addresses https://discord.com/channels/966627436387266600/1102510616326967306/1138143461908951111

Previously was only providing file key and url, now includes name and size as well 
to match with the input to `onUploadComplete()`

would be nice to have the objects be identical, however that would break back 
compat. 

`onUploadComplete()` uses this shape:

```
{
   url: string,
   key: string,
   name: string, 
   size: number
}
```

while `onClientUploadComplete()` uses:

```
{
   fileName: string,
   fileSize: number,
   fileKey: string,
   fileUrl: string
}
```

### CODE REVIEW

**Review by juliusmarminge — APPROVED** [review]
Maybe add a TODO to remove the `file` prefiz in the next major

File: packages/uploadthing/src/client.ts [hunk 1]
```diff
@@ -169,7 +169,10 @@ export const DANGEROUS__uploadFiles = async <TRouter extends FileRouter>(
     // Poll for file data, this way we know that the client-side onUploadComplete callback will be called after the server-side version
     await pollForFileData(presigned.key);
 
+    // TODO: remove `file` prefix in next major version
     return {
+      fileName: file.name,
+      fileSize: file.size,
       fileKey: presigned.key,
       fileUrl: genUrl,
```
Comment (juliusmarminge) [inline packages/uploadthing/src/client.ts:177]: Could also be a good idea to deprecate them now

```suggestion
    // @deprecated - use `name` instead
    fileName: file.name,
    name: file.name,
    // @deprecated - use `size` instead
    fileSize: file.size,
    size: file.size,
    // @deprecated - use `key` instead
    fileKey: presigned.key,
    key: presigned.key,
    // @deprecated - use `url` instead
    fileUrl: genUrl,
    url: genUrl,
```

File: packages/uploadthing/src/client.ts [hunk 2]
```diff
@@ -169,9 +169,20 @@ export const DANGEROUS__uploadFiles = async <TRouter extends FileRouter>(
     // Poll for file data, this way we know that the client-side onUploadComplete callback will be called after the server-side version
     await pollForFileData(presigned.key);
 
+    // TODO: remove `file` prefix in next major version
     return {
-      fileKey: presigned.key,
-      fileUrl: genUrl,
+    // @deprecated - use `name` instead
```
Comment (juliusmarminge) [inline packages/uploadthing/src/client.ts:174]: Indentation broke. Suggesting on mobile is hard 😅

Comment (markflorkowski) [reply packages/uploadthing/src/client.ts:174]: Haha I'll fix it in a minute.

**Comment by juliusmarminge** [comment]
hmm deprecated isn't showing up on hover from jsdoc :/ 
CleanShot 2023-08-08 at 08 55 56@2x

**Comment by juliusmarminge** [comment]
Got it:

CleanShot 2023-08-08 at 09 02 59@2x

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**